src/main/java/com/ericsson/ei/handlers/RmqHandler.java

```diff
@@ -148,45 +153,55 @@ public ConnectionFactory connectionFactory() {
     }
 
     @Bean
-    Queue queue() {
+    public Queue externalQueue() {
         return new Queue(getQueueName(), true);
     }
 
     @Bean
-    TopicExchange exchange() {
+    public Queue internalQueue() {
+        return new Queue(getWaitlistQueueName(), true);
+    }
+
+    @Bean
+    public TopicExchange exchange() {
         return new TopicExchange(exchangeName);
     }
 
     @Bean
-    Binding binding(Queue queue, TopicExchange exchange) {
-        return BindingBuilder.bind(queue).to(exchange).with(bindingKey);
+    Binding binding() {
+        return BindingBuilder.bind(internalQueue()).to(exchange()).with(WAITLIST_BINDING_KEY);
     }
 
     @Bean
-    public SimpleMessageListenerContainer bindToQueueForRecentEvents(ConnectionFactory springConnectionFactory,
+    public List<Binding> bindings() {
+        String[] bingingKeysArray = splitBindingKeys(bindingKeys);
+        List<Binding> bindingList = new ArrayList<Binding>();
+        for (String bindingKey : bingingKeysArray) {
+            bindingList.add(BindingBuilder.bind(externalQueue()).to(exchange()).with(bindingKey));
+        }
+        return bindingList;
+    }
+
+    @Bean
+    public SimpleMessageListenerContainer bindToQueueForRecentEvents(
```

Comment:
shouldn't this method have a javadoc?

Reply:
It seems to attach listeners to the queues? There are many public beans here without javadoc, should I type something on all of them? In master branch I think the code pretty much looks like this so that will be missing the javadoc.

Reply:
I would like them to be added, but consider it to be done on the master branch.